### ISSUE TITLE
Unexpected form submit when simulating {enter} despite preventDefault()

### ISSUE THREAD

**Issue by wkillerud** [testing-library/user-event#459]
Hi!

I've run into an issue with `userEvents.type()` when using `{enter}` inside a form. 

- `@testing-library/user-event` version: 12.1.6
- Testing Framework and version: Jest 26.4.2
- DOM Environment: JSDOM 16.2.2 through Jest

```javascript
userEvent.type(getByRole('textbox', { name: 'Destination' }), 'Mordor{enter}');
```

The scenario is simulating keyboard input in a combobox. The component highlights the first option that matches a search, and the user can hit Enter to select the option. To prevent a form submit the component calls `preventDefault()` on the keydown event. 

This works fine in a browser (tested in Firefox 81 and Chromium 85). The form is not submitted, as expected. However, in my test where I use `userEvents` the form is submitted on `{enter}`.

From what I can tell it should be a matter of adding the existing `keyDownDefaultNotPrevented` to the `if`-statement on [line 522 here](https://github.com/testing-library/user-event/blob/5feaa942f46bb37d96c2f2fbeb4b33e8beff75ad/src/type.js#L522)

```javascript
// handleEnter()

const keyDownDefaultNotPrevented = fireEvent.keyDown(currentElement(), {
    key,
    keyCode,
    which: keyCode,
    ...eventOverrides,
  })

// ...

if (
    currentElement().tagName === 'INPUT' &&
    currentElement().form &&
    (currentElement().form.querySelectorAll('input').length === 1 ||
      currentElement().form.querySelector('input[type="submit"]') ||
      currentElement().form.querySelector('button[type="submit"]'))
  ) {
    fireEvent.submit(currentElement().form)
  }
```

If the suggested solution sounds fine I can submit a PR. If you'd like a minimal reproduction repo I need some time extracting the minimum from the project I'm working on.

**Comment by ph-fritsche** [comment]
Will be solved by #437.

**Comment by malwilley** [comment]
#437 has been merged, this should no longer be an issue in the latest release:

https://github.com/testing-library/user-event/releases/tag/v12.1.7
